### PR TITLE
39 add instance segmentation to rf detr and yolo9

### LINKED ISSUE #39: Add Instance segmentation to RF-DETR and YOLO9
https://github.com/LibreYOLO/libreyolo/issues/30

### PR DESCRIPTION
**Pull request by EHxuban11**
Adds instance segmentation support for RF-DETR models across inference, training, ONNX export, and validation.           
  - New Masks class with data, xy/xyn contour properties, and cpu()/numpy() methods                                        
  - RF-DETR seg variants auto-detected from -seg filename suffix or segmentation_head keys in weights
  - Seg mask output at original image resolution, drawn as semi-transparent overlays under bounding boxes                  
  - ONNX export auto-detects seg models and exports 3 outputs (boxes, scores, masks)                                       
  - Polygon-format YOLO labels (segmentation datasets) auto-derive bounding boxes for training/validation                  
  - Tiled inference warns that masks are unsupported in tiled mode                                                         
  - rfdetr dependency bumped to >=1.6.0 (with [train] extra) for upstream seg config support                               
                                                                                                                           
Usage                                                                                                                                                                                                                                               
```
from libreyolo import LibreYOLO                                 
                                                                                                                           
model = LibreYOLO("LibreRFDETRn-seg.pt")                                                                                 
result = model("image.jpg", conf=0.3, save=True)                                                                         
                                                                                                                         
result.masks.data   # (N, H, W) bool tensor                                                                              
result.masks.xy     # list of (M, 2) pixel contour arrays                                                                
result.masks.xyn    # list of (M, 2) normalized contour arrays           
```                                                
                                                                                                                           
Test plan                                                                                                                                                                                                                            
  - Unit tests pass: pytest tests/unit/test_segmentation.py -v                                                             
  - E2E seg inference: pytest tests/e2e/test_rfdetr_seg_training.py::test_rfdetr_seg_inference_only -v
  - E2E seg training (CUDA): pytest tests/e2e/test_rfdetr_seg_training.py::test_rfdetr_seg_training -v                     
  - E2E resume from checkpoint (CUDA): pytest tests/e2e/test_rfdetr_seg_training.py::test_rfdetr_seg_resume_training -v    
  - Existing detection tests still pass (no regressions)                                                                   
  - ONNX export with seg model produces 3 outputs

### CODE REVIEW

File: libreyolo/data/yolo_coco_api.py [hunk 1]
```diff
@@ -47,21 +47,26 @@ def parse_yolo_label_line(
...
+        if len(parts) > 5:
+            # Segmentation format: class_id x1 y1 x2 y2 x3 y3 ...
+            # Derive bounding box from polygon vertices.
+            coords = [float(p) for p in parts[1:]]
+            xs = coords[0::2]
+            ys = coords[1::2]
+            cx = (min(xs) + max(xs)) / 2
+            cy = (min(ys) + max(ys)) / 2
+            bw = max(xs) - min(xs)
+            bh = max(ys) - min(ys)
```
Comment (brohoya) [inline libreyolo/data/yolo_coco_api.py:63]: This is duplicated here, it also exists in `dataset.py`.
Let's extract it to `data/utils.py`:
```python
def polygon_to_cxcywh(coords: list[float]) -> tuple[float, float, float, float]:
    """Derive normalized cxcywh bounding box from polygon vertices."""
    xs = coords[0::2]
    ys = coords[1::2]
    cx = (min(xs) + max(xs)) / 2
    cy = (min(ys) + max(ys)) / 2
    w = max(xs) - min(xs)
    h = max(ys) - min(ys)
    return cx, cy, w, h
```

File: libreyolo/backends/base.py [hunk 2]
```diff
@@ -253,7 +266,22 @@ def _parse_rfdetr(self, all_outputs, orig_w, orig_h, conf):
         boxes[:, [0, 2]] = np.clip(boxes[:, [0, 2]], 0, orig_w)
         boxes[:, [1, 3]] = np.clip(boxes[:, [1, 3]], 0, orig_h)
 
-        return boxes, max_scores, class_ids
+        # Resize and threshold masks to original image resolution
+        masks_out = None
+        if raw_masks is not None and len(raw_masks) > 0:
+            import torch
```
Comment (brohoya) [inline libreyolo/backends/base.py:272]: Function level import and already imported at the top of the file.

File: libreyolo/backends/base.py [hunk 3]
```diff
@@ -253,7 +266,22 @@ def _parse_rfdetr(self, all_outputs, orig_w, orig_h, conf):
...
+            import torch.nn.functional as F
```
Comment (brohoya) [inline libreyolo/backends/base.py:273]: Can use the top file torch import for this.

File: libreyolo/backends/base.py [hunk 4]
```diff
@@ -322,15 +362,22 @@ def _build_result(
 
     def _save_annotated(self, result, original_img, image_path, output_path):
         """Save annotated image to disk."""
+        annotated_img = original_img
         if len(result) > 0:
+            if result.masks is not None:
+                from ..utils.drawing import draw_masks
```
Comment (brohoya) [inline libreyolo/backends/base.py:368]: Let's move the `draw_masks` import at the top of the file where `utils.drawing` is already imported
